Sources/ArcGISToolkit/Components/Offline/OnDemand/OnDemandMapAreaSelectorView.swift

```diff
@@ -0,0 +1,310 @@
+    /// The corner radius of the area selector view.
+    private let cordnerRadius: CGFloat = 16
+    
+    /// The drag point for a drag gesture.
+    private enum DragPoint {
+        case topLeft, topRight, bottomLeft, bottomRight
+    }
+    
+    var body: some View {
+        GeometryReader { geometry in
+            ZStack {
+                ZStack {
+                    Rectangle()
+                        .fill(.black.opacity(0.10))
+                        .reverseMask {
+                            Rectangle()
+                                .frame(width: boundingRect.width, height: boundingRect.height)
+                                .position(boundingRect.center)
+                            
+                        }
+                    RoundedCorners(cornerRadius: cordnerRadius)
+                        .stroke(.ultraThickMaterial, style: StrokeStyle(lineWidth: 6, lineCap: .butt))
+                        .frame(width: insetRect.width, height: insetRect.height)
+                        .position(insetRect.center)
+                    
+                    ZStack {
+                        RoundedRectangle(cornerRadius: cordnerRadius, style: .continuous)
+                            .stroke(.white, lineWidth: 4)
+                        
+                    }
+                    .frame(width: boundingRect.width, height: boundingRect.height)
+                    .position(boundingRect.center)
+                }
+                .ignoresSafeArea()
+                .allowsHitTesting(false)
+                .onAppear {
+                    boundingRect = maxRect
+                    insetRect = boundingRect.insetBy(dx: -2, dy: -2)
+                    updateHandles()
+                }
+                .onChange(of: boundingRect) {
+                    selectedMapRect = $0
+                }
+                
+                Handle(position: topLeft) {
+                    resize(for: .topLeft, location: $0)
+                }
+                
+                Handle(position: topRight) {
+                    resize(for: .topRight, location: $0)
+                }
+                
+                Handle(position: bottomLeft) {
+                    resize(for: .bottomLeft, location: $0)
+                }
+                
+                Handle(position: bottomRight) {
+                    resize(for: .bottomRight, location: $0)
+                }
+            }
+            .onChange(of: safeAreaInsets) { _ in
+                let frame = CGRect(
+                    x: safeAreaInsets.leading,
+                    y: safeAreaInsets.top,
+                    width: geometry.size.width - safeAreaInsets.trailing - safeAreaInsets.leading,
+                    height: geometry.size.height - safeAreaInsets.bottom - safeAreaInsets.top
+                )
+                
+                maxRect = frame
+                    .insetBy(
+                        dx: frame.width * 0.1,
+                        dy: frame.height * 0.1
```

Comment:
I'd prefer hard-coded point values here instead of a percentage because a percentage will give different values for the width and height inset.